projects/hal-components/src/lib/components/paginator/paginator.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy, OnChanges, SimpleChanges, OnInit, ElementRef } from '@angular/core';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+
+import { Pagination } from './pagination';
+import { DEFAULT_PAGE_SIZE, DEFAULT_MAX_PAGE_SIZE } from './consts';
+import { SelectData } from '../selector/select-data.interface';
+
+@Component({
+  selector: 'hal-paginator',
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PaginatorComponent implements OnInit, OnChanges {
+  @Input() unit = 'elementer';
+  @Input() allowAll = true;
+  @Input() selectedPageIndex = 0;
+  @Input() length = 0;
+  @Input() pageSizeOptions = [10, 20];
+  @Input() selectedPageSize = DEFAULT_PAGE_SIZE;
+  @Input() maxPageSize = DEFAULT_MAX_PAGE_SIZE;
```

Comment:
Er ikke helt enig i navnet på denne 🤔  Kan virke som du setter hva største side/valg av pageSizeOptions som er lov. Hva med noe alla allOptionSize så skrive at den har en grense fortsatt, for å forhindre at du laster all data i universet på en gang